src/ucs/memory/memtype_cache.c

```diff
@@ -261,10 +275,14 @@ UCS_PROFILE_FUNC(ucs_status_t, ucs_memtype_cache_lookup,
         goto out_unlock;
     }
 
-    region      = ucs_derived_of(pgt_region, ucs_memtype_cache_region_t);
-    *mem_type_p = ((pgt_region->end >= (start + size)) ?
-                   region->mem_type : UCS_MEMORY_TYPE_LAST);
-    status      = UCS_OK;
+    if (ucs_likely((start + size) <= pgt_region->end)) {
+        region    = ucs_derived_of(pgt_region, ucs_memtype_cache_region_t);
+        *mem_info = region->mem_info;
+    } else {
+        mem_info->type    = UCS_MEMORY_TYPE_UNKNOWN;
+        mem_info->sys_dev = UCS_SYS_DEVICE_ID_UNKNOWN;
```

Comment:
maybe turn this initialization into a macro?

Reply:
added inline function